### ISSUE TITLE
ERROR: Connection failed. Server returned: Session reconnect failed: Could not find the specified session.

### ISSUE THREAD

**Issue by joyTQT** [sassoftware/python-dlpy#326]
Hi!

I am using DlPy to attempt retraining an RNN acoustic model for speech to text transcription.
When I run dlTrain on a collection of more than 25 audio files for table and validtable parameters each with a learning rate of 0.0001, I encounter the following error
![image](https://user-images.githubusercontent.com/70889006/92372433-7ceec080-f12f-11ea-9415-8a7766b6a0e1.png)

And with only 25 audio files, I get inaccurate transcriptions...

If I increase the learning rate to 0.001, I can take in 35 audio files for table and 25 for validtable but it does still encounter the error at times... 

Why does this error occur?
Is there any way to work around it?

Thanks!

**Comment by eusdougc** [comment]
@joyTQT  - are you retraining with a bare metal Viya installation or a containerized version of Viya?  The error you are seeing indicates that the Viya session has been killed or died.  We are seeing this kind of error more frequently with containerized Viya because some containers aren't configured with sufficient memory for deep learning.

**Comment by joyTQT** [comment]
@eusdougc - Hi! Thanks for the response!
You were right! It was an issue with insufficient memory for deep learning.

But is there a way to train in batches so that the session would not be terminated? 
e.g. train audio files 1-100, then 101-200, ...

**Comment by eusdougc** [comment]
@joyTQT  - the issue is generally not so much the size of the data but the size of the model.  When training we have to allocate space for gradients that are the size of the model for each thread.  If you are not specifying a value for nthreads when you invoke dlTrain, we use all available threads.  This tends to be very memory intensive.  I would suggest that you try specifying nthreads=1 when invoking dlTrain.  That is likely to be slow and inefficient, but if you don't lose your session, then you can try nthreads=5, nthreads=10, etc. until you hit a limit where you get the error you have been seeing.

**Comment by eusdougc** [comment]
@joyTQT - Have you tried my suggestion with nthreads?  If this solved your problem, are you OK with closing the issue?

**Comment by joyTQT** [comment]
Oh yeap! Lowering nThreads has solved my problem! Thank you!
